### ISSUE TITLE
Automatically update istio/istio dependencies more frequently

### ISSUE THREAD

**Issue by nschhina** [istio/istio.io#7708]
Right now if updates are done manually and only after every few weeks, the documentation testing might regress in istio.io and we wouldn't know until the istio/istio dependency is updated. This will take us very long to debug issues given the scope of changes that happen in istio/istio in matter of days. Instead we should leverage istio bot to do this automatically and once a day(hopefully).

**Comment by nschhina** [comment]
cc: @frankbu

**Comment by frankbu** [comment]
We've discussed this in the doc WG but so far have decided that it's best to only pick up a new istio/istio build periodically, when a new or updated doc needs it. The problem with picking up a new build every day is that failures caused by istio regressions/bugs cause a disruption to the doc process - it causes every PRs to fail, even though it has nothing to do with anything in the PR.

Using istio.io doc tests as a second level integration test for istio/istio doesn't work well, because it's too late, the PR causing problems is already merged. We could add the istio.io tests as part of the istio/istio test suite, but theoretically, istio/istio tests should have sufficient coverage of istio functionality themselves. The doc tests are really just to catch bugs in the docs (i.e., the documented instructions are still correct and don't need to be updated).

**Comment by frankbu** [comment]
Would it be possible to have istio/istio run the doc tests regularly (e.g., daily), as a validation step, so that istio.io will only ever pull verified builds for doc PR testing.

Alternatively, if we just pull istio/istio builds periodically, as needed, can we open a high priorty issue with a special label (e.g., doc-test-failure) that indicates an istio/istio regression that must be fixed asap?

**Comment by nschhina** [comment]
I think the second approach you suggested works better, a failing auto generated istio/istio dep update -> create high priority issue in istio/istio. This also doesn't block any future changes in istio/istio.io as a failed CI run wouldn't even allow to upgrade istio dep in istio.io.  cc: @howardjohn

**Comment by frankbu** [comment]
I guess we would need the second one, even if we also had the first. Having both would be ideal.

**Comment by howardjohn** [comment]
Why do you need the first if you have the second? Agree the second is good idea

**Comment by howardjohn** [comment]
And we have precedence - we do all sorts of cross repo updates today

**Comment by frankbu** [comment]
> Why do you need the first if you have the second? Agree the second is good idea

@howardjohn The thing about the first that I think is useful is that we would always (within a day) know if there is a regression in istio/istio that only the doc tests uncover, instead of letting them linger for a few days or weeks until we decide to upgrade the istio/istio build in istio.io. The longer we wait, the harder it is to track down what was changed that is causing the breakage (@nschhina's initial motivation for opening this PR).

I suppose we could achieve the same goal by automatically pulling a new istio/istio build daily, whether we need it or not, and using any failed PR, combined with the second suggestion (high priority label) to get it fixed, if necessary. 

The bottom line, is that unless the istio/istio testing is beefed up to the point that it covers everything that the doc testing does, there's value to running the doc tests regularly as an additional build verification for istio/istio.

**Comment by sdake** [comment]
@frankbu - agree. I don't see the harm in cross-repository gating. If there is cross-repository gating, no PR can ever break either build.

The concerns I heard were related to the time spent running the documentation test cases. The T&R infrastructure runs test cases in parallel, unless the test case is serial in nature. If the doc tests take some time to run (I have not heard a concrete number) - they could be split across tree boundaries.

Cheers,
-steve
